### DISCUSSION TITLE
大型电商网站的商品详情页系统架构中，nginx缓存问题

### DISCUSSION THREAD

**Discussion by Blusez** [doocs/advanced-java#200] [Q&A]
Nginx 本地缓存的数据是有一定的时间期限的，比如说 10 分钟，当数据过期之后，它就会从 redis 获取到最新的缓存数据，并且缓存到自己本地。
这样的话不就数据不一致了吗？而且最差会有10分钟这么长时间得不一致

**Comment by blackwhitechen** [comment]
不会吧

**Comment by Blusez** [comment]
> 不会吧

比如我nginx已经有本地缓存了，这时候，数据被更新了，有请求进来，本地缓存又没过期，那如何取到最新的数据呢

**Comment by huifer** [comment]
nginx 尽可能做静态资源的缓存, 为了防止nginx 缓存接口信息可以加上时间戳或者随机数让请求变成两个.
